hazelcast-client/src/main/java/com/hazelcast/client/connection/nio/ClientConnectionManagerImpl.java

```diff
@@ -364,19 +363,24 @@ protected ClientConnection createSocketConnection(final Address address) throws
         }
     }
 
+    private int getBufferSize() {
+        int bufferSize = socketOptions.getBufferSize() * KILO_BYTE;
+        if (bufferSize <= 0) {
+            bufferSize = DEFAULT_BUFFER_SIZE_BYTE;
+        }
+        return bufferSize;
+    }
+
     @Override
-    public void destroyConnection(final Connection connection, final String reason, final Throwable cause) {
+    public void onClose(Connection connection) {
         Address endPoint = connection.getEndPoint();
-        ClientConnection conn = (ClientConnection) connection;
-        if (endPoint != null && connections.remove(endPoint, conn)) {
+
+        if (endPoint != null && connections.remove(endPoint, connection)) {
             logger.info("Removed connection to endpoint: " + endPoint + ", connection: " + connection);
 
-            conn.close(reason, cause);
-            for (ConnectionListener connectionListener : connectionListeners) {
-                connectionListener.connectionRemoved(conn);
+            for (ConnectionListener listener : connectionListeners) {
+                listener.connectionRemoved(connection);
             }
-        } else {
-            connection.close(reason, cause);
```

Comment:
Nobody else than the connection will call the 'onClose'. So the connection is being closed.